store/store.go

```diff
@@ -1228,6 +1240,40 @@ var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo)
 	return targetSnapPath, nil
 }
 
+// downloadAndApplyDelta downloads and then applies the delta to the current snap.
+func (s *Store) downloadAndApplyDelta(name, targetFn string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (err error) {
+	deltaInfo := &downloadInfo.Deltas[0]
+	workingDir := filepath.Dir(targetFn)
+
+	deltaPath, err := s.downloadDelta(name, workingDir, downloadInfo, pbar, user)
+	if err != nil {
+		return err
+	}
+	defer os.Remove(deltaPath)
+
+	logger.Debugf("Successfully downloaded delta for %q at %s", name, deltaPath)
+	snapFromDeltaPath, err := applyDelta(name, deltaPath, deltaInfo)
+	if err != nil {
+		return err
+	}
+	defer os.Remove(snapFromDeltaPath)
+
+	bsha3_384, _, err := osutil.FileDigest(snapFromDeltaPath, crypto.SHA3_384)
+	if err != nil {
+		return err
+	}
+	sha3_384 := fmt.Sprintf("%x", bsha3_384)
+	if downloadInfo.Sha3_384 != "" && sha3_384 != downloadInfo.Sha3_384 {
+		return fmt.Errorf("sha3-384 mismatch after patching %s: got %s but expected %s", name, sha3_384, downloadInfo.Sha3_384)
```

Comment:
s/%s/%q/ for the name, as it will be just a plain word. The hashes are fine as %s.